### PR TITLE
NVIDIA boards

### PR DESCRIPTION
**Pull request by pattacini**
This PR addresses #326.

This is the final outcome.

Screenshot 2024-01-04 142412

> [!warning]
Quantities underlined in `#ff0000` need specific review since I didn't manage to find them out or I'm uncertain about. Essentially, the RAM and the Disk space.

> [!note]
I've reported the carrier boards for the form factor. Feel free to propose alternatives.

### CODE REVIEW

File: docs/icub_cpu_boards/icub_cpu_boards.md [hunk 1]
```diff
@@ -10,5 +10,8 @@ The table below reports all CPU boards installed on iCub.
...
 | COMEXPRESS Advantech |   |  ADVANTECH_SOM-7583C7-S8A1  |   Advantech  |  COM-Express Type10, Advantech SOM i7-1185G7E 16GB DDR4, 64GB SSD                                  |  Com-Express Type 10    | ergoCub 1.x           | Intel i7-1185G7E 4 cores 2.8GHz  |  16GB RAM DDR4  |  onboard 64GB NVMe SSD |  | 
+| Xavier NX | | Jetson Xavier NX | NVIDIA® | NVIDIA Jetson Xavier NX Series (21 TOPS) | Connect Tech Quark Carrier Board | iCub 2.10, iCub 3.0, iRonCub 3.0 | 384-core Volta GPU, 6-core Carmel Arm 64-bit CPU | 16GB (8GB) RAM | | |
+| Xavier AGX | | Jetson Xavier AGX | NVIDIA® | NVIDIA Jetson Xavier AGX Series (32 TOPS) | Connect Tech Rogue Carrier Board | ergoCub 1.x | 512-core Volta GPU, 8-core Carmel Arm 64-bit CPU | 16GB RAM | | |
+| Orin NX | | Jetson Orin NX | NVIDIA® | NVIDIA Jetson Orin NX Series (50 TOPS) | Connect Tech Boson for FRAMOS Carrier Board | ergoCub 1.1 | 1024-core Ampere GPU, 8-core Arm Cortex-A78AE 64-bit CPU | 16GB RAM | 64GB SSD | |
```
Comment (maggia80) [inline docs/icub_cpu_boards/icub_cpu_boards.md:15]: external M.2 SSD

File: docs/icub_cpu_boards/icub_cpu_boards.md [hunk 2]
```diff
@@ -10,5 +10,8 @@ The table below reports all CPU boards installed on iCub.
 | COMEXPRESS CONGATEC  |   | conga-TC87/i7-4650U     | Congatec       | COM-Express Type6, Congatec TC87/i7-4650U (Core i7-4650U 1.7GHz) p/n 46901                           | COM-Express Type6   | iCub 2.5, iCub 3.0        | Core i7-4650U             | 8GB       | external SATA 32GB  |               |
 | COMEXPRESS KONTRON   |   | COMe-cKL6i7-7600 8GB32S | Kontron        | COM-Express Type6, Kontron COMe-cKL6i7-7600 8GB RAM onboard and 32GB SLC eMMC 5.0 (Core i7-7600U 1.7GHz) | COM-Express Type6   | iCub 2.5-2.6 and 2.7  | Intel i7-7600U            | 8+8GB RAM | onboard 32GB        |               |
 | COMEXPRESS Advantech |   |  ADVANTECH_SOM-7583C7-S8A1  |   Advantech  |  COM-Express Type10, Advantech SOM i7-1185G7E 16GB DDR4, 64GB SSD                                  |  Com-Express Type 10    | ergoCub 1.x           | Intel i7-1185G7E 4 cores 2.8GHz  |  16GB RAM DDR4  |  onboard 64GB NVMe SSD |  | 
+| Xavier NX | | Jetson Xavier NX | NVIDIA® | NVIDIA Jetson Xavier NX Series (21 TOPS) | Connect Tech Quark Carrier Board | iCub 2.10, iCub 3.0, iRonCub 3.0 | 384-core Volta GPU, 6-core Carmel Arm 64-bit CPU | 16GB (8GB) RAM | | |
```
Comment (maggia80) [inline docs/icub_cpu_boards/icub_cpu_boards.md:13]: 16GB eMMC

File: docs/icub_cpu_boards/icub_cpu_boards.md [hunk 3]
```diff
@@ -10,5 +10,8 @@ The table below reports all CPU boards installed on iCub.
...
+| Xavier AGX | | Jetson Xavier AGX | NVIDIA® | NVIDIA Jetson Xavier AGX Series (32 TOPS) | Connect Tech Rogue Carrier Board | ergoCub 1.x | 512-core Volta GPU, 8-core Carmel Arm 64-bit CPU | 16GB RAM | | |
```
Comment (maggia80) [inline docs/icub_cpu_boards/icub_cpu_boards.md:14]: 32 GB eMMC + external M.2 SSD

**Review by maggia80 — COMMENTED** [review]
I have added the info about eMMC and SSD support for the 3 configurations

**Review by traversaro — APPROVED** [review]

File: docs/icub_cpu_boards/icub_cpu_boards.md (hunk 3 above)
Comment (pattacini) [reply docs/icub_cpu_boards/icub_cpu_boards.md]: Done!

File: docs/icub_cpu_boards/icub_cpu_boards.md (hunk 2 above)
Comment (pattacini) [reply docs/icub_cpu_boards/icub_cpu_boards.md]: Done!

File: docs/icub_cpu_boards/icub_cpu_boards.md (hunk 1 above)
Comment (pattacini) [reply docs/icub_cpu_boards/icub_cpu_boards.md]: Done!

**Comment by pattacini** [comment]
> I have added the info about eMMC and SSD support for the 3 configurations

Thanks heaps @maggia80 👍🏻 
I've incorporated your comments!

The only missing point is the RAM capacity of Xavier NX. Is it 16GB or 8GB?

**Review by Nicogene — APPROVED** [review]

**Comment by maggia80** [comment]
> acity of Xavier NX. Is it 16GB or 8GB?

16GB

**Comment by pattacini** [comment]
Thanks! Fixed via the latest commit.
We can merge it.

**Pull request merged**